tools/cli/adaparse.cpp

```diff
@@ -255,15 +244,14 @@ int main(int argc, char** argv) {
   }
 
   if (result.count("path")) {
-    std::string filepath = result["path"].as<std::string>();
-    FILE* file = fopen(filepath.c_str(), "r");
+    auto file_path = result["path"].as<std::string>();
+    auto file = fopen(file_path.c_str(), "r");
```

Comment:
It is debatable whether hiding the type by using `auto` is a net gain. It makes writing the code faster, but it can make reading it harder.

I don't mind here or in general,  but it is worth thinking about.